datajunction-server/datajunction_server/internal/nodes.py

```diff
@@ -839,33 +857,42 @@ def _create_node_from_inactive(
         if previous_inactive_node.type != new_node_type:
             raise DJException(  # pragma: no cover
                 message=f"A node with name `{data.name}` of a `{previous_inactive_node.type.value}` "  # pylint: disable=line-too-long
-                "type existed before. If you want to re-created with a different type now, "
-                "you need to remove all the traces of the previous node with a <TODO> command.",
+                "type existed before. If you want to re-create it with a different type, "
+                "you need to remove all traces of the previous node with a hard delete call: "
+                "DELETE /nodes/{node_name}/hard",
```

Comment:
Have we changed our position on clients freely using the hard delete endpoint? I don't have any problems if so but if we want to keep that as a server admin utility then we shouldn't mention it in this error response.

Reply:
@samredai re: materialization, I was thinking that as a next step I could add notifications, which would then make it so that in the scheduling background task, we can catch exceptions and add it to the notifications queue. 

re: hard delete: good point, I guess we can keep this message and just block if someone tries to call that endpoint and doesn't have permissions. Although I think users could judiciously use this for nodes that they own, so as long as we check for ownership, maybe we can just allow for hard deletes?

Reply:
If we ask for a confirmation on hard deletes, then I don't see why hiding this feature. People should be able to delete their nodes if they so desire, unless EY consultants tell us not to ;)